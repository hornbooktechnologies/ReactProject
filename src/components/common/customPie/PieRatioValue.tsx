import { useTranslation } from 'react-i18next'
import { getDiffOfValues, IsappearTooltip } from '../../../utils/utils';
import Tooltip from '../Tooltip';

type Props = {
    forecastHrs: number
    forecastpreviousHrs: number
    planHrs: number
    planpreviousHrs: number
    actual: boolean
    suffix:string
}

const PieRatioValue = (props: Props) => {
    const { forecastHrs, actual, planHrs,forecastpreviousHrs,planpreviousHrs,suffix} = props;
    const { i18n } = useTranslation()
    let difForecast = getDiffOfValues(forecastHrs,forecastpreviousHrs)
    let diffPlan =  getDiffOfValues(planHrs,planpreviousHrs)
     
    return (
      <>
        <div className="uk-text-center">
          <div className="uk-flex uk-flex-center uk-flex-middle uk-flex-column uk-margin-auto-left uk-margin-auto-right">
            <div className="uk-border-dashed uk-width-medium">
              <h3 className="uk-margin-remove font-20 mb-5">
                <span className={`${IsappearTooltip(difForecast)}`}>
                  {forecastHrs}
                  <span className="font-8 uk-text-normal ml-3">{suffix}</span>
                {difForecast === true ? (
                  <Tooltip
                    previousDayValue={forecastpreviousHrs}
                    unit={suffix}
                  />
                ) : null}
                </span>
              </h3>
              <p className="font-12 mt-0 mb-10">
                {actual ? i18n.t('LABELS.ACTUAL') : i18n.t('LABELS.FORECAST')}
              </p>
            </div>
            <div className="uk-width-medium">
              <p className="font-12 mt-10 mb-0">{i18n.t('LABELS.PLAN')}</p>
              <h3
                className="uk-margin-remove font-20">
                <span className={`${IsappearTooltip(diffPlan)}`}>
                {planHrs}
                  <span className="font-8 uk-text-normal ml-3">{suffix}</span>
                {diffPlan === true ? (
                  <Tooltip previousDayValue={planpreviousHrs} unit={suffix} />
                ) : null}
                 </span>
              </h3>
            </div>
          </div>
        </div>
      </>
    )
}

export default PieRatioValue