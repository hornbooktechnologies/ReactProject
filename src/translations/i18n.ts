import { initReactI18next } from 'react-i18next'
import i18n from 'i18next'
import LanguageDetector from 'i18next-browser-languagedetector'
import { TRANSLATIONS_EN } from './en/Translations'
import { TRANSLATIONS_JA } from './ja/Translations'

i18n
  .use(LanguageDetector)
  .use(initReactI18next)
  .init({
    lng: 'en',
    resources: {
      en: {
        translation: TRANSLATIONS_EN,
      },
      ja: {
        translation: TRANSLATIONS_JA,
      },
    },
  })

export default i18n
