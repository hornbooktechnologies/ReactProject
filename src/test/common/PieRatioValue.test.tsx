import React from "react";
import PieRatioValue from "../../components/common/customPie/PieRatioValue";
import renderer from "react-test-renderer";
import i18n from "../../i18nForTests";
import { I18nextProvider } from "react-i18next";
describe("Component: PieRationValue", () => {
  var Props = {
    forecastHrs: 10,
    forecastpreviousHrs: 12,
    planHrs: 15,
    planpreviousHrs: 14,
    actual: true,
    suffix: "oku",
  };

  it("PieRationValue renders correctly", () => {
    const tree = renderer
      .create(
        <I18nextProvider i18n={i18n}>
          <PieRatioValue {...Props} />
        </I18nextProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
