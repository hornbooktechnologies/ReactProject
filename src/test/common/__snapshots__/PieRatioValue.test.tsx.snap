// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: PieRationValue PieRationValue renders correctly 1`] = `
<div
  className="uk-text-center"
>
  <div
    className="uk-flex uk-flex-center uk-flex-middle uk-flex-column uk-margin-auto-left uk-margin-auto-right"
  >
    <div
      className="uk-border-dashed uk-width-medium"
    >
      <h3
        className="uk-margin-remove font-20 mb-5"
      >
        <span
          className="uk-tooltip-bottom-custom uk-tooltip-cutom text-highlight"
        >
          10
          <span
            className="font-8 uk-text-normal ml-3"
          >
            oku
          </span>
          <div
            className="uk-transdorm-origin-center-bottom uk-tooltip-bottom-center uk-transform-origin-center-bottom uk-tooltip uk-animation-fade-up uk-animation-center uk-togglabe-enter uk-text-left font-10 uk-text-center font-normal"
          >
            <span
              className="color-grey-blue-80 uk-display-block"
            >
              UNIT_OPERATION.PREVIOUS_DAY
            </span>
            <span
              className="font-10 uk-text-bold uk-display-block mt-5"
            >
              12
               
              oku
            </span>
          </div>
        </span>
      </h3>
      <p
        className="font-12 mt-0 mb-10"
      >
        LABELS.ACTUAL
      </p>
    </div>
    <div
      className="uk-width-medium"
    >
      <p
        className="font-12 mt-10 mb-0"
      >
        LABELS.PLAN
      </p>
      <h3
        className="uk-margin-remove font-20"
      >
        <span
          className="uk-tooltip-bottom-custom uk-tooltip-cutom text-highlight"
        >
          15
          <span
            className="font-8 uk-text-normal ml-3"
          >
            oku
          </span>
          <div
            className="uk-transdorm-origin-center-bottom uk-tooltip-bottom-center uk-transform-origin-center-bottom uk-tooltip uk-animation-fade-up uk-animation-center uk-togglabe-enter uk-text-left font-10 uk-text-center font-normal"
          >
            <span
              className="color-grey-blue-80 uk-display-block"
            >
              UNIT_OPERATION.PREVIOUS_DAY
            </span>
            <span
              className="font-10 uk-text-bold uk-display-block mt-5"
            >
              14
               
              oku
            </span>
          </div>
        </span>
      </h3>
    </div>
  </div>
</div>
`;
